components/gantry/oneaxis/oneaxis_test.go

```diff
@@ -42,14 +43,29 @@ func createFakeMotor() motor.Motor {
 	}
 }
 
-func createFakeBoard() board.Board {
+func createLimitBoard() board.Board {
 	injectGPIOPin := &inject.GPIOPin{
 		GetFunc: func(ctx context.Context, extra map[string]interface{}) (bool, error) { return true, nil },
 		SetFunc: func(ctx context.Context, high bool, extra map[string]interface{}) error { return nil },
 	}
 	return &inject.Board{GPIOPinByNameFunc: func(pin string) (board.GPIOPin, error) { return injectGPIOPin, nil }}
 }
 
+func createFakeBoard() board.Board {
+	injectGPIOPin := &inject.GPIOPin{
+		GetFunc: func(ctx context.Context, extra map[string]interface{}) (bool, error) {
+			pinCount := time.Now().Unix()
+			if pinCount%2 == 0 {
```

Comment:
Not sure about how I'm doing this. If I create a `pinCount` variable outside this function I get a race condition because I'm incrementing and accessing the same variable across multiple coroutines. I can foresee this potentially failing the timeout if it never happens to get the time on an even number but I find that unlikely. Is this acceptable?

Reply:
This function is only ever called from one coroutine, right? I wouldn't expect any race conditions from that. You're right that a time-based implementation is nondeterministic and has the potential to get messy. So, I'd make the variable local within `createFakeBoard`:
```go
func createFakeBoard() board.Board {
    pinCounter := 0
    injectGPIOPin := &inject.GPIOPin{
        GetFunc: func(ctx context.Context, extra map[string]interface{}) (bool, error) {
            pinCounter += 1
            return (pinCounter % 10 == 0), nil
        }
    }
```

Reply:
missed this before. changing now but it seems to be hanging and I'm not sure why